### ISSUE TITLE
Enemy Display Glitch

### ISSUE THREAD

**Issue by rnveach** [bibendovsky/bstone#477]
Using version 1.2.12 , sometimes the enemies appear glitched out. This seems to happen more with the Drones and Volatile Transports. It is like it isn't sure what texture to use and where for these enemies and randomly cycles through the possibilities on each side of the enemy for each frame.

Video: https://www.youtube.com/watch?v=ojxllZXiY9k (initial video shows current graphic options, bug occurs between 0:25 and 0:30)
Save: https://rveach.no-ip.org/bstone_aog_full_saved_game_7

**Comment by bibendovsky** [comment]
It's not a glitch.
Here is a video for reference from the original game.

[bs_aog_e4f2_psd_and_vt_anims.webm](https://github.com/bibendovsky/bstone/assets/1597301/663c25ac-851f-4596-a186-d698ef5a5abe)

**Comment by rnveach** [comment]
Thank you. Do you mean your saying this isn't a glitch because the original game also produced the same behavior, so it is there to maintain "compatibility"? The reason I thought this was a glitch was because this doesn't happen all the time. There are levels were these same enemies don't exhibit this behavior and seem what I would consider "normal" without the weird rotation.

**Comment by bibendovsky** [comment]
> Thank you. Do you mean your saying this isn't a glitch because the original game also produced the same behavior, so it is there to maintain "compatibility" with the original release? The reason I thought this was a glitch was because this doesn't happen all the time. There are levels were these same enemies don't exhibit this behavior and seem what I would consider "normal" without the weird rotation.

Yes, the port tries to be compatible as much as possible with original games.

You could observe such behavior on any walking/running/chasing enemy stuck (obstructed by other enemy or player) for a while in the tile.

On the other hand, this continuous spinning of the floating drone until it detects the player, could be really a glitch.

**Comment by rnveach** [comment]
Ok, i wasn't sure what direction this port was going, but its nice to know it gives a priority to being as same to the original as possible .

Just my opinion, but a switch feature could be added for this to stabilize the enemy display. It won't have any meaningful negative impact on the game from the original and atleast improve their display so they don't look like they are bugging out .

Otherwise, I understand that this issue can be closed if this will be left as it is.
